### ISSUE TITLE
Консоль запросов

### ISSUE THREAD

**Issue by zeegin** [marmyshev/edt-plugins#12]
Консоль, которая может открываться из перспективы отладки в момент подключенного предмета отладки.

В консоли можно собирать запрос в конструкторе, текстом, выполнять запрос в менеджере запросов, просматривать запросы менеджера, изменять параметры запроса, исполнять запрос и смотреть результат.

При выполнении запроса запускать сборку логов техжурнала для анализа плана запроса.

**Comment by marmyshev** [comment]
Было бы классно ещё план запроса посмотреть - по выполненному запросу

**Comment by marmyshev** [comment]
Надо бы научить 1С-предприятие - чтобы умело самостоятельно перезапускать обработку

**Comment by zeegin** [comment]
Ну план запроса по конретному запросу выцепить не сложно, пример есть в консоли БСП конфигурирования техжурнала.

**Comment by zeegin** [comment]
А зачем нужна обработка?
Ведь исполнить запрос можно через команды dbgs.

**Comment by marmyshev** [comment]
Ну я подозреваю, что dbgs просто так ничего не сможет выполнить, если не будет «точки останова» на сервере. 

к тому же нужно создать менеджер менеджер временных таблиц или найти где-то существующий... 

чтобы сделать это «универсально» как-то, я думал что можно запускать спец.обработку в которую ставить точку останова на сервере и подключаться - дальше делать всё автоматом... 

либо тогда уж «просить пользователя» остановиться в нужном месте и указать существующего менеджера временных таблиц

**Comment by zeegin** [comment]
Я считаю что переход к консоли должен быть где то рядом с панелью вотч. 
Или из ктрл+ф9 на объекте запрос. 
А в этот момент сервер будет в режиме отладки и на брейепоинте как раз

**Comment by marmyshev** [comment]
Да, тогда уже сейчас можно такой плагин сделать - всё для этого есть уже...

 «бери и делай!» (с)

**Comment by VitaliyVS-2020** [comment]
> Я считаю что переход к консоли должен быть где то рядом с панелью вотч.
> Или из ктрл+ф9 на объекте запрос.
> А в этот момент сервер будет в режиме отладки и на брейепоинте как раз

Делал плагин с подобным функционалом,  dbgs спокойно вычисляет выражения, переданные одной строкой, но для полной свободы нужна возможность выполнения нескольких строк кода и фиксация в переменных промежуточных результатов. Такое возможно даже не реализуемо в dbgs.

**Comment by kalapuc10** [comment]
Было бы круто, если бы был executor в редакторе запроса EDT в любом месте.
